components/brave_wallet/browser/zcash/zcash_orchard_storage.cc

```diff
@@ -0,0 +1,445 @@
+        !meta_table.SetVersionNumber(kCurrentVersionNumber)) {
+      NOTREACHED_IN_MIGRATION();
+      database_->Close();
+      return false;
+    }
+  }
+
+  return true;
+}
+
+bool ZCashOrchardStorage::CreateSchema() {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+
+  sql::Transaction transaction(database_.get());
+  return transaction.Begin() &&
+         database_->Execute("CREATE TABLE " kNotesTable
+                            " "
+                            "("
+                            "id INTEGER PRIMARY KEY AUTOINCREMENT,"
+                            "account_id TEXT NOT NULL,"
+                            "amount INTEGER NOT NULL,"
+                            "block_id INTEGER NOT NULL,"
+                            "nullifier BLOB NOT NULL UNIQUE);") &&
+         database_->Execute("CREATE TABLE " kSpentNotesTable
+                            " "
+                            "("
+                            "id INTEGER PRIMARY KEY AUTOINCREMENT,"
+                            "account_id TEXT NOT NULL,"
+                            "spent_block_id INTEGER NOT NULL,"
+                            "nullifier BLOB NOT NULL UNIQUE);") &&
+         database_->Execute("CREATE TABLE " kAccountMeta
+                            " "
+                            "("
+                            "account_id TEXT NOT NULL PRIMARY KEY,"
+                            "account_birthday INTEGER NOT NULL,"
+                            "latest_scanned_block INTEGER NOT NULL,"
+                            "latest_scanned_block_hash TEXT NOT NULL);") &&
+         transaction.Commit();
+}
+
+bool ZCashOrchardStorage::UpdateSchema() {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  return true;
+}
+
+std::optional<ZCashOrchardStorage::Error> ZCashOrchardStorage::RegisterAccount(
+    mojom::AccountIdPtr account_id,
+    uint64_t account_birthday_block,
+    const std::string& account_birthday_block_hash) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+
+  if (!EnsureDbInit()) {
+    NOTREACHED_IN_MIGRATION();
+    return Error{kDbInitError, "Failed to init database "};
+  }
+
+  sql::Transaction transaction(database_.get());
+  if (!transaction.Begin()) {
+    NOTREACHED_IN_MIGRATION();
+    return Error{kDbInitError, "Failed to init database "};
+  }
+
+  sql::Statement register_account_statement(database_->GetCachedStatement(
+      SQL_FROM_HERE, "INSERT INTO " kAccountMeta " "
+                     "(account_id, account_birthday, latest_scanned_block, "
+                     "latest_scanned_block_hash) "
+                     "VALUES (?, ?, ?, ?)"));
+
+  register_account_statement.BindString(0, account_id->unique_key);
+  register_account_statement.BindInt64(1, account_birthday_block);
+  register_account_statement.BindInt64(2, account_birthday_block);
+  register_account_statement.BindString(3, account_birthday_block_hash);
+
+  if (!register_account_statement.Run()) {
+    transaction.Rollback();
```

Comment:
No need in `transaction.Rollback();` just before return (here and below).

Reply:
Done